manager/manager.go

```diff
@@ -775,35 +778,24 @@ func (m *manager) createContainer(containerName string) error {
 		return err
 	}
 
-	// Add to the containers map.
-	alreadyExists := func() bool {
-		m.containersLock.Lock()
-		defer m.containersLock.Unlock()
-
-		namespacedName := namespacedContainerName{
-			Name: containerName,
-		}
-
-		// Check that the container didn't already exist.
-		_, ok := m.containers[namespacedName]
-		if ok {
-			return true
```

Comment:
The problem IIUC is that of not invoking Cleanup() before letting go GC the container object. 
I intended to say that if `alreadyExists` is `true`, then we can invoke Cleanup() right away.
But I prefer not creating the object if it already exists and invoking Cleanup() iif we return error instead of starting housekeeping. 
That would let us not add the new `Start` method.